### PR TITLE
UX-652 Convert Spinner to ts, Specifies width and height system props

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Converts `Spinner` to typescript
- Spinner was accepting all system props, it is now restricted to `margin, position, width, height`
- Removes a `@ts-ignore` in the Button component, fixes Button eslint errors

### PR Checklist

- [x] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -137,155 +140,146 @@ export type BaseProps = {
...
-          {loadingIndicator}
-        </StyledButton>
-      );
-    }
+type PolymorphicButton = Polymorphic.ForwardRefComponent<
+  'button',
+  BaseProps & MarginProps & WidthProps & PaddingProps
+>;
+
+const Button = React.forwardRef(function Button(props, ref) {
```
Comment (jonambas) [inline packages/matchbox/src/components/Button/Button.tsx:150]: added `function Button` here for the displayname eslint error

File: .eslintrc [hunk 2]
```diff
@@ -20,7 +20,7 @@
         "ignoreRestSiblings": true
       }
     ],
-    "react/prop-types": 1
+    "react/prop-types": "off"
```
Comment (jonambas) [inline .eslintrc:23]: This was causing false positives with `forwardRef` components

File: libby/feedback/Spinner.lib.tsx [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import { describe, add } from '@sparkpost/libby-react';
-import { Spinner, Box } from '@sparkpost/matchbox';
+// @ts-ignore
+import { Box } from '@sparkpost/matchbox';
+import { Spinner } from '../../packages/matchbox/src';
```
Comment (logansparlin) [inline libby/feedback/Spinner.lib.tsx:5]: Can change to `import { Spinner } from '@sparkpost/matchox'` if you merge in main

File: packages/matchbox/src/components/Spinner/Spinner.tsx [hunk 4]
```diff
@@ -52,15 +69,6 @@ const Spinner = React.forwardRef(function Spinner(props, ref) {
...
-  color: PropTypes.oneOf(['gray', 'orange', 'blue', 'white']),
-  label: PropTypes.string.isRequired,
-  rotationOnly: PropTypes.bool,
-  ...createPropTypes(margin.propNames),
-  ...createPropTypes(position.propNames),
-};
-
 Spinner.defaultProps = {
   size: 'medium',
 };
```
Comment (logansparlin) [inline packages/matchbox/src/components/Spinner/Spinner.tsx:74]: could just do `size = 'medium'` in props spread and remove this?

**Review by logansparlin — COMMENTED** [review]
Looks good, just a couple small things

**Pull request merged**